src/pubd/repository.rs

```diff
@@ -653,12 +687,37 @@ impl RrdpServer {
         let notification_path_new = self.notification_path_new();
         let notification_path = self.notification_path();
         self.notification.write_xml(&notification_path_new)?;
-        fs::rename(notification_path_new, notification_path)?;
+        fs::rename(&notification_path_new, &notification_path).map_err(|e| {
+            KrillIoError::new(
+                format!(
+                    "Could not rename notification file from '{}' to '{}'",
+                    notification_path_new.to_string_lossy(),
+                    notification_path.to_string_lossy()
+                ),
+                e,
+            )
+        })?;
 
         // clean up under the base dir:
         // - old session dirs
-        for entry in fs::read_dir(&self.rrdp_base_dir)? {
-            let entry = entry?;
+        for entry in fs::read_dir(&self.rrdp_base_dir).map_err(|e| {
+            KrillIoError::new(
+                format!(
+                    "Could not read RRDP base directory '{}'",
+                    self.rrdp_base_dir.to_string_lossy()
+                ),
+                e,
+            )
+        })? {
+            let entry = entry.map_err(|e| {
+                KrillIoError::new(
+                    format!(
+                        "Could not read RRDP base directory '{}'",
```

Comment:
This error message is the same as the one above. Should something be added to make them distinct?

Reply:
ok changed this to "Could not read entry in RRDP base directory '{}'"